device.go

```diff
@@ -152,3 +152,16 @@ func (d *device) CommitUpdate() error {
 	// For now set only appropriate boot flags
 	return d.WriteEnv(BootVars{"upgrade_available": "0"})
 }
+
+func (d *device) HasUpdate() (bool, error) {
```

Comment:
Reordered and squashed a number of commits (possibly blowing up github review in the process).